docs/src/main/asciidoc/property-source-config.adoc

```diff
@@ -597,6 +602,44 @@ https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Configuration-Bi
 You can find an example of an application that uses secrets (though it has not been updated to use the new `spring-cloud-kubernetes` project) at
 https://github.com/fabric8-quickstarts/spring-boot-camel-config[spring-boot-camel-config]
 
+[[namespace-resolution]]
+=== Namespace resolution
+Finding an application namespace happens on a best-effort basis. There are some steps that we iterate in order
+to find it. The easiest and most common one, is to specify it in the proper configuration, for example:
+
+====
+[source,yaml]
+----
+spring:
+  application:
+    name: app
+  cloud:
+    kubernetes:
+      secrets:
+        name: secret
+        namespace: default
+        sources:
+         # Spring Cloud Kubernetes looks up a Secret named 'a' in namespace 'default'
+         - name: a
+         # Spring Cloud Kubernetes looks up a Secret named 'secret' in namespace 'b'
+         - namespace: b
+         # Spring Cloud Kubernetes looks up a Secret named 'd' in namespace 'c'
+         - namespace: c
+           name: d
+----
+====
+
+Remember that the same can be done for config maps. If such a namespace is not specified, it will be read (in this order):
+
```

Comment:
properly document the order in which resolution of a namespace happens. Also add a clear line that failure to do will result in an Exception.